xdsl/ir.py

```diff
@@ -15,12 +15,31 @@
 OpT = TypeVar('OpT', bound='Operation')
 
 
+@dataclass
+class Dialect:
+    """Contains the operations and attributes of a specific dialect"""
+    _operations: list[Operation] = field(default_factory=list,
+                                         init=True,
+                                         repr=True)
+    _attributes: list[Attribute] = field(default_factory=list,
+                                         init=True,
+                                         repr=True)
+
+
 @dataclass
 class MLContext:
     """Contains structures for operations/attributes registration."""
     _registeredOps: dict[str, type[Operation]] = field(default_factory=dict)
     _registeredAttrs: dict[str, type[Attribute]] = field(default_factory=dict)
 
+    def register_dialect(self, dialect: Dialect):
+        """Register a dialect. Operation and Attribute names should be unique"""
```

Comment:
Can this be verified somehow?

Reply:
This is verified within the call to `register_op`. Would you prefer to verify it before calling as well?

Reply:
ok than, great